src/test/SecurityPanel/StoreJmxCredentials.test.tsx

```diff
@@ -0,0 +1,263 @@
+import userEvent from '@testing-library/user-event';
+import renderer, { act } from 'react-test-renderer';
+import { render, screen } from '@testing-library/react';
+import { of } from 'rxjs';
+import { Target } from '@app/Shared/Services/Target.service';
+
+import '@testing-library/jest-dom';
+import { StoreJmxCredentials } from '@app/SecurityPanel/StoreJmxCredentials';
+import { ServiceContext, defaultServices } from '@app/Shared/Services/Services';
+import { Modal, ModalVariant } from '@patternfly/react-core';
+import { NotificationMessage } from '@app/Shared/Services/NotificationChannel.service';
+
+const mockTarget: Target = { connectUrl: 'service:jmx:rmi://someUrl', alias: 'fooTarget' };
+const mockAnotherTarget: Target = { connectUrl: 'service:jmx:rmi://anotherUrl', alias: 'anotherTarget' };
+
+jest.mock('@app/SecurityPanel/CreateJmxCredentialModal', () => {
+  return {
+    CreateJmxCredentialModal: jest.fn((props) => {
+      return (
+        <Modal
+          isOpen={props.visible}
+          variant={ModalVariant.large}
+          showClose={true}
+          onClose={props.onClose}
+          title="CreateJmxCredentialModal"
+        >
+          Jmx Auth Form
+        </Modal>
+      );
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/NotificationChannel.service', () => {
+  const mockNotification = { message: { target: 'service:jmx:rmi://someUrl' } } as NotificationMessage;
+  return {
+    ...jest.requireActual('@app/Shared/Services/NotificationChannel.service'),
+    NotificationChannel: jest.fn(() => {
+      return {
+        messages: jest
+          .fn()
+          .mockReturnValueOnce(of()) // 'renders correctly'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'adds the correct table entry when a stored notification is received'
+          .mockReturnValueOnce(of())
+          .mockReturnValueOnce(of(mockNotification))
+
+          .mockReturnValueOnce(of()) // 'removes the correct table entry when a deletion notification is received'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValueOnce(of()) // 'renders an empty table after receiving deletion notifications for all credentials'
+          .mockReturnValueOnce(of(mockNotification))
+          .mockReturnValueOnce(of())
+
+          .mockReturnValue(of()), // all other tests
+      };
+    }),
+  };
+});
+
+jest.mock('@app/Shared/Services/Target.service', () => {
+  return {
+    ...jest.requireActual('@app/Shared/Services/Target.service'),
+    TargetService: jest.fn(() => {
+      return {
+        deleteCredentials: jest.fn(),
+      };
+    }),
+  };
+});
+
+describe('<StoreJmxCredentials />', () => {
+  beforeAll(() => {
+    jest.resetModules();
+    const ApiService = require('@app/Shared/Services/Api.service').default;
+    const TargetsService = require('@app/Shared/Services/Api.service').default;
```

Comment:
The import path for `TargetsService` is incorrect here (the error is still the same though after fixing this).